optuna/integration/mlflow.py

```diff
@@ -0,0 +1,135 @@
+import optuna
+from optuna._experimental import experimental
+from optuna import structs
+from optuna import type_checking
+
+if type_checking.TYPE_CHECKING:
+    from typing import Dict  # NOQA
+    from typing import Optional  # NOQA
+
+try:
+    import mlflow
+
+    _available = True
+except ImportError as e:
+    _import_error = e
+    _available = False
+    mlflow = object
+
+
+def _check_mlflow_availability():
+    # type: () -> None
+
+    if not _available:
+        raise ImportError(
+            "MLflow is not available. Please install MLflow to use this "
+            "feature. It can be installed by executing `$ pip install "
+            "mlflow`. For further information, please refer to the installation guide "
+            "of MLflow. (The actual import error is as follows: " + str(_import_error) + ")"
+        )
+
+
+@experimental("1.4.0")
+class MLflowCallback(object):
+    """Callback to track optuna trials with MLflow.
+
+    This callback adds relevant information that is
+    tracked by Optuna to MLflow. You must set 'study.study_name' inside
+    Optuna because it is needed to set the MLflow experiment name.
+
+    Example:
+
+        Add MLflow callback to optuna optimization.
+
+        .. testsetup::
+
+            import pathlib
+            import tempfile
+            tempdir = tempfile.mkdtemp()
+            YOUR_TRACKING_URI = pathlib.Path(tempdir).as_uri()
+
+        .. testcode::
+
+            import optuna
+            from optuna.integration.mlflow import MLflowCallback
+
+            def objective(trial):
+                x = trial.suggest_uniform('x', -10, 10)
+                return (x - 2) ** 2
+
+            mlflc = MLflowCallback(
+                tracking_uri=YOUR_TRACKING_URI,
+                metric_name='my metric score',
+            )
+
+            study = optuna.create_study(study_name='my_study')
+            study.optimize(objective, n_trials=10, callbacks=[mlflc])
+
+        .. testcleanup::
+
+            import shutil
+            shutil.rmtree(tempdir)
+
+        .. testoutput::
+            :hide:
+            :options: +NORMALIZE_WHITESPACE
+
+            INFO: 'my_study' does not exist. Creating a new experiment
+
+    Args:
+        tracking_uri:
+            The URI of the MLflow tracking server.
+
+            Please refer to `mlflow.set_tracking_uri
+            <https://www.mlflow.org/docs/latest/python_api/mlflow.html#mlflow.set_tracking_uri>`_
+            for more details.
+        metric_name:
+            Name of the metric. If not provided this will be called ``trial_value``.
+    """
+
+    def __init__(self, tracking_uri=None, metric_name=None):
```

Comment:
You don't have to hide the default argument value behind and a `None` and could use the actual `"value"`. Then you could simplify the documentation of this parameter omitting the explanation about the default value, the type hint and also the logic of checking whether it's a `None`.
```suggestion
    def __init__(self, tracking_uri=None, metric_name="value"):
```

Reply:
Ok. Changed this.